Level.py

```diff
@@ -1,15 +1,210 @@
+# coding: utf8
+
 from kivy.uix.floatlayout import FloatLayout
+from kivy.graphics import Color, Point
 from MapCanvas import MapCanvas
 
 
 class Level(FloatLayout):
     def __init__(self, map_file_path, textures, **kwargs):
         """
         Charge la carte dans un layout.
+        Charge les données et la matrix de tile et de touch
         :param map_file_path: Chemin de la carte
         :param textures: dictionnaire de textures
         :param kwargs: Argumens du layout
         """
         super(Level, self).__init__(**kwargs)
         self.map_canvas = MapCanvas(map_file_path, textures)
         self.add_widget(self.map_canvas)
+
+        # charge les informations de touch et de tile
+        self.touch_width = int
+        self.touch_scaling_factor = 6
+
+        self.level_size = list
+        self.tile_size = list
+
+        self.xmax = int
+        self.ymax = int
+        self.matrix = None
+
+        self.color = (1, 1, 1)
+        self.old_point = list
+        self.point = list
+
+        self.define_level_properties()
+
+        # load level victory path
+        self.player_path = []
+        self.win_path = []
+        self.get_win_conditions()
+
+    """
+    level properties for initialisation
+    """
+
+    def define_level_properties(self):
+        # defini l'épaisseur de trait de touch
+        self.touch_width = self.map_canvas.tile_size / self.touch_scaling_factor
+
+        # defini la taille reel des tile
+        self.level_size = [self.map_canvas.window.size[0] - self.map_canvas.vectical_padding * 2,
+                           self.map_canvas.window.size[1] - self.map_canvas.horizontal_padding * 2]
+        self.tile_size = [self.level_size[0] / self.map_canvas.map_size[0],
+                          self.level_size[1] / self.map_canvas.map_size[1]]
+
+        # initialise la matrice
+        self.xmax = self.map_canvas.map_size[0]
+        self.ymax = self.map_canvas.map_size[1]
+        self.matrix = [[0 for _ in xrange(self.xmax)] for _ in xrange(self.ymax)]
+
+        # rempli la matrice avec les coordonnées
+        x = self.map_canvas.vectical_padding
+        y = self.map_canvas.window.size[1] - self.map_canvas.horizontal_padding
+
+        for indx in range(self.ymax):
```

Comment:
index_x corrected, don't understand question, index_x is the current index between 0 and x_max